src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -2612,9 +2620,28 @@ def update_public_ip(cmd, instance, dns_name=None, allocation_method=None, versi
         instance.sku.name = sku
     if ip_tags:
         instance.ip_tags = ip_tags
+    if public_ip_prefix:
+        SubResource = cmd.get_models('SubResource')
+        instance.public_ip_prefix = SubResource(id=public_ip_prefix)
     return instance
 
 
+def create_public_ip_prefix(cmd, client, resource_group_name, public_ip_prefix_name, prefix_length,
+                            location=None, tags=None):
+    PublicIPPrefix, PublicIPPrefixSku = cmd.get_models('PublicIPPrefix', 'PublicIPPrefixSku')
+    prefix = PublicIPPrefix(
+        location=location,
+        prefix_length=prefix_length,
+        sku=PublicIPPrefixSku(name='Standard'),
```

Comment:
should we expose the sku as an argument, or `standard` is the only valid value so far?

Reply:
Right. It is the only valid value.  If you don't supply it, ironically the service defaults to the unsupported "Basic" and throws an error 🤷‍♂️